File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool descending(int a, int b)
{
    return a > b;
}

int solution(vector<int> citations) {
    sort(citations.begin(), citations.end(), descending);
    
    int index = 0;
    for (int i = 1; i <= citations.size(); ++i)
    {
        if (citations[i - 1] >= i)
            index = i;
        else
            break;
    }
    
    return index;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort(reverse=True)  # 인용 횟수를 내림차순으로 정렬
    h_index = 0

    for i, c in enumerate(citations, start=1):  # 1부터 시작하는 인덱스로 순회
        if c >= i:  # 인용 횟수가 현재 순위(i)보다 크거나 같으면 H-Index 가능
            h_index = i
        else:
            break  # 더 이상 조건을 만족하지 않으면 중단

    return h_index